src/enum.lisp

```diff
@@ -37,59 +47,138 @@
 ;;; constant for the type by a type translator when passed as
 ;;; arguments or a return value to a foreign function.
 
-(defclass foreign-enum (foreign-typedef enhanced-foreign-type)
+(defclass foreign-enum (named-foreign-type enhanced-foreign-type)
   ((keyword-values
-    :initform (make-hash-table :test 'eq)
+    :initform (error "Must specify KEYWORD-VALUES.")
+    :initarg :keyword-values
     :reader keyword-values)
    (value-keywords
-    :initform (make-hash-table)
+    :initform (error "Must specify VALUE-KEYWORDS.")
+    :initarg :value-keywords
     :reader value-keywords))
   (:documentation "Describes a foreign enumerated type."))
 
-(defun make-foreign-enum (type-name base-type values)
-  "Makes a new instance of the foreign-enum class."
-  (let ((type (make-instance 'foreign-enum :name type-name
-                             :actual-type (parse-type base-type)))
-        (default-value 0))
+(deftype enum-key ()
+  '(and symbol (not null)))
+
+(defparameter +valid-enum-base-types+ '(:char :unsigned-char
+                                        :short :unsigned-short
+                                        :int :unsigned-int
+                                        :long :unsigned-long
+                                        :long-long :unsigned-long-long))
+
+(defun parse-foreign-enum-like (type-name base-type values
+                                &optional field-mode?)
```

Comment:
I know this is bikeshedding, but CFFI uses the, admittedly retro, `-p` convention. :)